File: programmers/합성수 찾기.js
// 약수의 개수가 세 개 이상인 수를 합성수라고 합니다.
// 자연수 n이 매개변수로 주어질 때 n이하의 합성수의 개수를 return

function solution(n) {
  // 중복없도록
  let set = new Set();

  for (let num = 1; num <= n; num++) {
    for (let i = 2; i < num; i++) {
      if (num % i === 0) {
        //set이라 push 대신 add
        set.add(num);
      }
    }
  }

  // set이라 length 대신 size
  return set.size;
}

console.log(solution(10) === 5);

File: programmers/python/합성수 찾기.py
# 합성수 찾기
def solution(n):
    answer = n
    for i in range(1, n + 1):
        for j in range(2, i):
            if i % j == 0:
                answer -= 1
                break
    return n - answer


def solution(n):
    output = 0
    for i in range(4, n + 1):
        for j in range(2, int(i ** 0.5) + 1):
            if i % j == 0:
                output += 1
                break
    return output